cpp/src/IceGrid/DescriptorBuilder.h

```diff
@@ -18,11 +18,10 @@ namespace IceGrid
         XmlAttributesHelper(const XMLAttributes&, const Ice::LoggerPtr&, const std::string&, int);
 
         void checkUnknownAttributes();
-        bool contains(const std::string&) const;
-        std::map<std::string, std::string> asMap() const;
-
-        bool asBool(const std::string&) const;
-        bool asBool(const std::string&, bool) const;
+        bool contains(const std::string&) const; // NOLINT:modernize-use-nodiscard
```

Comment:
All these functions update the mutable _used. In some code, we call these functions only to update _used. And yes, that's not very elegant.